web/js/codeworld_shared.js

```diff
@@ -260,42 +287,48 @@ function registerStandardHints(successFunc)
         // Filter out CallStack constraints.
         line = line.replace(/:: HasCallStack =>/g, '::');
 
-        var wordStart = 0;
-        if (line.startsWith("type ") || line.startsWith("data ")) {
-            wordStart += 5;
-
-            // Hide kind annotations.
-            var kindIndex = line.indexOf(" ::");
-            if (kindIndex != -1) {
-                line = line.substr(0, kindIndex);
+        if (line.startsWith("-- ")) {
```

Comment:
To be technically correct, I think what you want is:

    if (line.startsWith("-- |")) {
      doc = ...
    } else if (doc != "" && line.startsWith("-- ")) {
      doc += ...
    }

Reply:
True. Updating.